submissions/ses3332002/tiny-js-world-oop/index.js

```diff
@@ -0,0 +1,97 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Being {
+  constructor (name, friends, species) {
+    this.species = species;
+    this.name = name;
+    this.friends = friends;
+  }
+  
+  getDescription() {
+    return Object.entries(this)
+    .filter(([key]) => propsToPrint.includes(key))
+    .map(([,value]) => Array.isArray(value) ? value.join(', ') : value)
+    .join('; ');
+  }
+};
+
+class Animal extends Being {
+  constructor (name, friends, gender, saying, species) {
+    super(name, friends, species);
+    this.gender = gender;
+    this.saying = saying;
+    this.legs = 4;
+    this.hands = 0;
+  }
+};
+
+class Dog extends Animal {
+  constructor (name, friends, gender) {
+    super(name, friends, gender);
```

Comment:
Why not sending species over to the constructor of the parent class via `super`?

Reply:
Specie of an instances of the Dog class is always "dog". Why send it via the constructor?

Reply:
It would be a valid question if i requested you to pass species name at construction time like `new Dog("dog", ...)`

"Why to send a shared property via constructor?" would be a more appropriate question here.
Reason is that `species` is a shared property so it has to be handled by a class that owns this property.

This way child classes let parent class (owner) to handle the property as it wishes.

We have a simple and short structure here. This type of things become critically important on big projects. It's better to learn it now while no one is punished for mistakes.

Check OOP and SOLID principles [here](https://github.com/OleksiyRudenko/a-tiny-JS-world#learn-on-your-own) and employ them to write some better code that is maintainable and scalable.

Reply:
I remade it.